flax/nn/normalization.py

```diff
@@ -89,27 +89,28 @@ def apply(self,
                          'use_running_averages is True')
       mean, var = ra_mean.value, ra_var.value
     else:
-      mean = jnp.mean(x, axis=reduction_axis, keepdims=True)
+      mean = jnp.mean(x, axis=reduction_axis, keepdims=False)
       if axis_name is not None and not self.is_initializing():
-        axis_size = lax.psum(1., axis_name=axis_name)
-        mean = lax.psum(mean, axis_name=axis_name) / axis_size
+        mean = lax.pmean(mean, axis_name=axis_name)
 
-      mean2 = jnp.mean(lax.square(x), axis=reduction_axis, keepdims=True)
+      mean2 = jnp.mean(lax.square(x), axis=reduction_axis, keepdims=False)
       if axis_name is not None and not self.is_initializing():
-        mean2 = lax.psum(mean2, axis_name=axis_name) / axis_size
+        mean2 = lax.pmean(mean2, axis_name=axis_name)
       var = mean2 - lax.square(mean)
 
       if ra_mean and not self.is_initializing():
         ra_mean.value = momentum * ra_mean.value + (1 - momentum) * mean
         ra_var.value = momentum * ra_var.value + (1 - momentum) * var
 
-    y = x - mean
+    y = x - mean.reshape(feature_shape)
```

Comment:
shouldn't this be `reduced_feature_shape`?  this is what's causing test breakage I think.

Reply:
Forgot to copy over a line; should be working now.